src/features/UpdatePowerShell.ts

```diff
@@ -16,15 +16,27 @@ const PowerShellGitHubPrereleasesUrl =
 
 export class GitHubReleaseInformation {
     public static async FetchLatestRelease(preview: boolean): Promise<GitHubReleaseInformation> {
+        const requestConfig: RequestInit = {};
+
+        // For CI. This prevents GitHub from rate limiting us.
+        if (process.env.PS_TEST_GITHUB_API_USERNAME && process.env.PS_TEST_GITHUB_API_PAT) {
+            const authHeaderValue = Buffer
+                .from(`${process.env.PS_TEST_GITHUB_API_USERNAME}:${process.env.PS_TEST_GITHUB_API_PAT}`)
+                .toString("base64");
+            requestConfig.headers = {
+                Authorization: `Basic ${authHeaderValue}`,
+            };
+        }
+
         // Fetch the latest PowerShell releases from GitHub.
         let releaseJson: any;
         if (preview) {
             // This gets all releases and the first one is the latest prerelease if
             // there is a prerelease version.
-            releaseJson = (await fetch(PowerShellGitHubPrereleasesUrl)
+            releaseJson = (await fetch(PowerShellGitHubPrereleasesUrl, requestConfig)
```

Comment:
Isn't the point of `await` to obviate the need for `then()`?

Reply:
It's either this or double awaiting which I felt was a tad less clear with the extra parens and less verbose than an intermediate variable. I can change it if you'd like.

```js
await (await fetch(PowerShellGitHubReleasesUrl, requestConfig)).json()

// or

const res = await fetch(PowerShellGitHubReleasesUrl, requestConfig);
releaseJson = await res.json();
```

Reply:
No no, no need to derail the PR